src/paper/tasks.py

```diff
@@ -1308,107 +1307,38 @@ def pull_new_openalex_works(page=0, retry=0):
-                concepts = open_alex.hydrate_paper_concepts(raw_concepts)
-
-                # if paper exists, we try to update it
-                if existing_paper:
-                    # we update these specific fields because we also create new papers from arxiv
-                    # and arxiv usually doesn't have these fields
-                    existing_paper.paper_publish_date = data.get("paper_publish_date")
-                    existing_paper.alternate_ids = data.get("alternate_ids", {})
-                    existing_paper.citations = data.get("citations")
-                    if existing_paper.abstract is None:
-                        existing_paper.abstract = data.get("abstract")
-                    if data.get("pdf_license") is not None:
-                        existing_paper.pdf_license = data.get("pdf_license")
-                    if data.get("oa_status") is not None:
-                        existing_paper.oa_status = data.get("oa_status")
-                    existing_paper.is_open_access = data.get("is_open_access")
-                    existing_paper.open_alex_raw_json = data.get("open_alex_raw_json")
-
-                    existing_paper.save()
-                    paper_id = existing_paper.id
-
-                    # update concepts
-                    create_paper_concepts_and_hubs.apply_async(
-                        (
-                            paper_id,
-                            concepts,
-                        ),
-                        priority=2,
-                        countdown=1,
-                    )
+            if not cursor or not works or len(works) == 0:
+                break
 
-                    num_papers_success += 1
-                    continue
-
-                # create paper
-                paper = Paper(**data)
-                paper.full_clean()
-                paper.get_abstract_backup()
-                paper.get_pdf_link()
-                paper.save()
-                paper_id = paper.id
-
-                # we're purposefully skipping PDF download here,
-                # since there's on the order of new 30-40K papers/day
-                # and that's just too much data processing for us to handle
-                # download_pdf.apply_async((paper_id,), priority=3, countdown=5)
-
-                # trigger add orcid authors
-                add_orcid_authors.apply_async((paper_id,), priority=5, countdown=5)
-
-                # create concepts and hubs
-                create_paper_concepts_and_hubs.apply_async(
-                    (
-                        paper_id,
-                        concepts,
-                    ),
-                    priority=2,
+            for work in works:
+                _process_openalex_work.apply_async(
+                    (work,),
+                    priority=4,
                     countdown=1,
                 )
+                # delay so that we don't queue too many tasks at once
+                time.sleep(0.2)
```

Comment:
It's slightly different, countdown delays the execution of the task, whereas `time.sleep` delays the queuing.

I added it so that we wouldn't just be constantly fetching works and adding them to the queue right away, this `time.sleep` is meant to give the queue some time to run through the queued tasks.

I don't know how efficacious it'd even be tbh, but I wanted to add something to reduce load.